### PR TITLE
Feature/add button component

### LINKED ISSUE #6: FE: Create button
Use headlessui as a base, https://headlessui.com/

Button should have options to set predefined sizes and a liquid size option. Liquid size option makes the button expand the entire width of its parent's width.

### PR DESCRIPTION
**Pull request by AmirAgassi**
🔘 **Add Button Component**

🔍 **What's Included**
- Add `Button` component.

📁 Files Affected:
- `src/components/Button.tsx`
- `src/components/index.ts`
- `src/pages/Landing.tsx`

### CODE REVIEW

File: src/components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }, ref) => {
+        // base styles that are always applied
+        const baseStyles = 'rounded-md font-medium transition-colors duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 disabled:opacity-50 disabled:cursor-not-allowed m-2';
+        
+        // variant styles
+        const variantStyles = {
+            primary: 'bg-gray-700 text-white hover:bg-gray-800 focus:ring-gray-500',
+            secondary: 'bg-gray-100 text-gray-700 hover:bg-gray-200 focus:ring-gray-400',
+            outline: 'border-2 border-gray-700 text-gray-700 hover:bg-gray-50 focus:ring-gray-400'
+        };
```
Comment (juancwu) [inline src/components/Button.tsx:37]: I can see how this pattern of assigning different classes will be in almost every component, we can implement https://cva.style/docs/getting-started/installation later on to make this cleaner and easier to manage.

**Review by juancwu — APPROVED** [review]
ltgm 👍

File: src/components/Button.tsx (hunk 1 above)
Comment (AmirAgassi) [reply src/components/Button.tsx:37]: > I can see how this pattern of assigning different classes will be in almost every component, we can implement https://cva.style/docs/getting-started/installation later on to make this cleaner and easier to manage.

How have I never heard of this?? For sure

**Review by SherRao — CHANGES_REQUESTED** [review]
these are shit

**Comment by AmirAgassi** [comment]
> these are shit

ur fired buddy

**Pull request merged**